frontend/public/components/sidebars/resource-sidebar-samples.tsx

```diff
@@ -0,0 +1,284 @@
+        templateName: 'read-write-deployment-in-ext-and-apps-apis',
+        kind: referenceForModel(RoleModel),
+      },
+      {
+        title: 'Allow different access rights to different types of resource and API groups',
+        description:
+          'This "Role" is allowed to read "Pods" and read/write "Jobs" resources in API groups.',
+        templateName: 'read-pods-and-read-write-jobs',
+        kind: referenceForModel(RoleModel),
+      },
+      {
+        title: 'Allow reading a ConfigMap in a specific namespace (for RoleBinding)',
+        description:
+          'This "Role" is allowed to read a "ConfigMap" named "my-config" (must be bound with a "RoleBinding" to limit to a single "ConfigMap" in a single namespace).',
+        templateName: 'read-configmap-within-ns',
+        kind: referenceForModel(RoleModel),
+      },
+      clusterRoleBinding1,
+      clusterRoleBinding2,
+    ],
+  )
+  .setIn([referenceForModel(ClusterRoleModel)], [clusterRoleBinding1, clusterRoleBinding2]);
+
+export const getResourceSidebarSamples = (kindObj: K8sKind, yamlSampleObj: K8sResourceKind) => {
+  const yamlSampleData = !_.isEmpty(yamlSampleObj)
+    ? _.filter(
+        yamlSampleObj.data,
+        (sample) => sample.spec.apiVersion === apiVersionForModel(kindObj),
+      )
+    : [];
+  const yamlSamples = [];
+  _.each(yamlSampleData, (sample) => {
+    yamlSamples.push(_.get(sample, 'spec'));
+  });
+  const existingSamples = defaultSamples.get(referenceForModel(kindObj));
+  return !_.isEmpty(existingSamples) ? existingSamples.concat(yamlSamples) : yamlSamples;
+};
+
+const ResourceSidebarSample: React.FC<ResourceSidebarSampleProps> = ({
+  sample,
+  loadSampleYaml,
+  downloadSampleYaml,
+}) => {
+  const { highlightText, title, img, description, templateName, yaml, kind } = sample;
+  return (
+    <li className="co-resource-sidebar-item">
+      <h3 className="h4">
+        <span className="text-uppercase">{highlightText}</span> {title}
+      </h3>
+      {img && <img src={img} className="co-resource-sidebar-item__img img-responsive" />}
+      <p>{description}</p>
+      <Button variant="link" isInline onClick={() => loadSampleYaml(templateName, yaml, kind)}>
+        <PasteIcon className="co-icon-space-r" />
+        Try it
+      </Button>
+      <Button
+        variant="link"
+        isInline
+        className="pull-right"
+        onClick={() => downloadSampleYaml(templateName, yaml, kind)}
+      >
+        <DownloadIcon className="co-icon-space-r" />
+        Download YAML
+      </Button>
+    </li>
+  );
+};
+
+export const ResourceSidebarSamples: React.FC<ResourceSidebarSamplesProps> = ({
+  samples,
+  loadSampleYaml,
+  downloadSampleYaml,
+}) => {
+  return (
+    <ol className="co-resource-sidebar-list">
+      {_.map(samples, (sample, i) => (
+        <ResourceSidebarSample
+          key={sample.templateName + i || sample.title + i}
```

Comment:
I might just use `i` if we're going to do this. We could consider adding an id to `Sample` that we can use (with the uid of the custom sample and just the name for the default samples)

Reply:
Reading React docs for hack day and ran across https://www.npmjs.com/package/shortid via https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318.  Seems like a winner to me.